File: 02/cb.py
def string_times(str, n):
  result = ""
  for i in range(n):  
    result = result + str  
  return result

def front_times(str, n):
  if len(str) < 3:
    front = str
  else:
    front = str[:3]
  result = ""
  for i in range(n):
    result += front
  return result

def string_bits(str):
  result = ""
  for i in range(len(str)):
    if i % 2 == 0:
     result = result + str[i]
  return result

def lone_sum(a, b, c):
  if a == b == c:
    return 0
  elif a == b:
    return c
  elif a == c:
    return b
  elif b == c:
    return a
  else:
    return a + b + c

def string_splosion(str):
  result = ""
  for i in range(len(str)+1):
    result += str[:i]
  return result

def cigar_party(cigars, is_weekend):
  if is_weekend:
    if cigars>=40:
      return True
    else:
      return False
  else:
    if cigars>=40 and cigars<=60:
      return True
    else:
      return False

def caught_speeding(speed, is_birthday):
  gift = 0
  if is_birthday:
    gift = 5
  if speed <= 60+gift:
    return 0
  elif speed >= 81+gift:
    return 2
  else:
    return 1

print (string_times('Hi',2))
print (string_times('Hi', 3))
print (string_times('Hi', 1))

print (front_times('Chocolate', 2))
print (front_times('Chocolate', 3))
print (front_times('Abc', 3))

print (string_bits('Hello'))
print (string_bits('Hi'))
print (string_bits('Heeololeo'))

print (lone_sum(1, 2, 3))
print (lone_sum(3, 2, 3)) 
print (lone_sum(3, 3, 3))

print (string_splosion('Code'))
print (string_splosion('abc')) 
print (string_splosion('ab'))

print (cigar_party(30, False))
print (cigar_party(50, False)) 
print (cigar_party(70, True))

print (caught_speeding(60, False))
print (caught_speeding(65, False)) 
print (caught_speeding(65, True))

File: 13/cb.cpp
#include <iostream>
int lone_sums (int a, int b, int c){
	if (a == b == c){
		std::cout << 0;
	} else if (a == b){
		std::cout << c;
	} else if (b == c){
		std::cout << a;
	} else if (a == c){
		std::cout << b;
	} return a + b + c;
}
int cigars_party(int cigars,bool is_weekend){
	if (is_weekend){
			if (cigars >= 40){
			return  true;
		}return false;
		}else{
			if (cigars >= 40 and cigars <= 60){
				return true;
			return false;
			}
		}

	}
int caught_speeding (int speed, bool is_birthday){
	  int gift;
	  gift = 0;
  if (is_birthday){
  	int gift = 5;
  }
    
  if (speed <= 60+gift){
  	return 0;
  }
    
  else if (speed >= 81+gift){
    return 2;
} return 1;
}




int main(){
	int a = 1;
		int b = 2;
		int c = 3;
	int x;
	x = lone_sums(a,b,c);
	std::cout << x << std::endl;

	int cigars = 30;
		 bool is_weekend = false;
		bool z;
		z = cigars_party(cigars, is_weekend);
		std::cout << z << std::endl;

	int speed = 60;
	bool is_birthday = false;
	int y;
	y = caught_speeding(speed, is_birthday);
	std::cout << y << std::endl;

	return 0;
}
 

